feat: add jamstack arch (#946)
aziontech/docs: src/content/docs/en/pages/architectures/jamstack/jamstack.mdx

---
title: Deploy Jamstack Websites at the Edge
description: Use Azion and Jamstack to improve websites' capabilities with resilient, secure, and scalable solutions.

> meta_tags: architecture, build, bot, development, jamstack

Reviewer: ```suggestion
meta_tags: architecture, build, development, jamstack
```

namespace: docs_arch_jamstack_deploy
permalink: /documentation/architecture/jamstack/deploy-jamstack-applications/
---

Jamstack website architectures leverage pre-rendered pages and APIs to design web applications, significantly improving e-commerce capabilities with resilient, secure, and scalable solutions. These architectures manage every part of the application, from front-end to back-end, using APIs running on JavaScript, reusable Markup templates, and prebuilt static Assets, resulting in enhanced performance, higher security, cost-effectiveness, and a superior developer experience. Meanwhile, leveraging edge computing—processing data at the network's edge closest to the user—can reduce latency and bandwidth use, further boosting performance.

This solution is ideal for organizations looking to enhance website performance, improve the overall user experience, and empower developers to rapidly build and deploy high-performance, scalable applications at the edge, closer to end-users.

---

## Jamstack Website Architecture Diagram

![Architecture diagram](/assets/docs/images/uploads/jamstack-deploy.png)

### Jamstack Website Dataflow 

> 1. The developer interacts with Azion's platform using tools like the web-based console and the command-line interface (CLI). These tools are essential for configuring and managing applications and services on the platform.

Reviewer: ```suggestion
1. The developer interacts with Azion's platform using tools like the web-based Console and the command-line interface (CLI). These tools are essential for configuring and managing applications and services on the platform.
```

6. The output from the edge node is delivered to the end-users via a domain, representing the user-facing endpoint of the web where the hosted application is accessible.

### Components

- [Azion Console](/en/documentation/products/guides/getting-to-know-azion-console/): the interface where you can access all the available products and their settings.
- [Azion CLI](/en/documentation/products/azion-cli/overview/): an open-source command-line interface (CLI) that allows interaction with Azion Edge Platform using a terminal. 
- [Edge Application](/en/documentation/products/build/edge-application/): an Azion product that allows you to build applications that run in Azion's edge network.
- [Edge Storage](/en/documentation/products/store/edge-storage/): a scalable and secure storage service designed to integrate object storage with the Azion Edge Platform using the S3 standard for object operations.

---

## Implementation 

- Initialize a Jamstack Edge Application based on templates:

> [Framework specific guides]

Reviewer: ```suggestion
[Framework specific guides](/en/documentation/products/build/develop-with-azion/frameworks-specific/overview/)
```

- Link an existing application to an Azion edge application:

> [Azion CLI link command]

Reviewer: ```suggestion
[Azion CLI link command](/en/documentation/products/cli/link-command/)
```

---

> title: Deployment of Jamstack Websites at the Edge

Reviewer: ```suggestion
title: Deploy Jamstack Websites at the Edge
```

> description: Improving websites capabilities with resilient, secure, and scalable solutions with Azion and Jamstack.

Reviewer: ```suggestion
description: Use Azion and Jamstack to improve websites' capabilities with resilient, secure, and scalable solutions.
```

Reviewer: the other archs have a similar structure here

meta_tags: architecture, build, bot, development, jamstack
namespace: docs_arch_jamstack_deploy
permalink: /documentation/architecture/jamstack/deploy-jamstack-applications/
---

> Jamstack website architectures leverage pre-rendered pages and APIs to design web applications, significantly improving e-commerce capabilities with resilient, secure, and scalable solutions. These architectures manage every part of the application, from front-end to back-end, using APIs running on JavaScript, reusable Markup templates, and prebuilt static Assets, resulting in enhanced performance, higher security, cost-effectiveness, and a superior developer experience. Meanwhile, leveraging edge computing—processing data at the network's edge closest to the user—can reduce latency and bandwidth use, further boosting performance.

Reviewer: remember to link to Lnfing Page

This solution is ideal for organizations looking to enhance website performance, improve the overall user experience, and empower developers to rapidly build and deploy high-performance, scalable applications at the edge, closer to end-users.

---

## Jamstack Website Architecture Diagram

![Architecture diagram](/assets/docs/images/uploads/jamstack-deploy.png)

> 

Reviewer: ```suggestion
```

### Jamstack Website Dataflow 

1. The developer begins by interacting with the Azion platform using tools like the web-based console and the command-line interface (CLI). These tools are essential for configuring and managing applications and services on the platform.
2. Developers work on application code locally on their machines. When ready, they commit changes to their local Git repository and then push these commits to a remote repository. In the case of working with Azion, this remote repository could be linked to Azion's platform to trigger further actions.
3. The applications files are sent to Edge Storage, then a bucket is created and the files are stored in it. The developer can store data and application assets here, which can be processed quickly by the edge application.
4. Azion Actions sets up a Continuous Deployment flow, connecting the GitHub repository with the Azion edge application created in the deployment. This process is possible through the communication between Azion Actions and Azion API.

> 5. Azion Actions deploy the edge application and its cofigurations, based on a default config or the one informed by the user in a manifest file. The configs include Rules Engine, Edge Cache, Image Processor, and Edge Functions

Reviewer: ```suggestion
5. Azion Actions deploy the edge application and its cofigurations, based on a default config or the one informed by the user in a manifest file. The configs include Rules Engine, Edge Cache, Image Processor, and Edge Functions.
```

6. The output from the edge node is delivered to the end-users via a domain, representing the user-facing endpoint of the web where the hosted application is accessible.

> ---
> 
> ## Components 
> 
> ### Components

Reviewer: ```suggestion
---
### Components
```

> - [Azion Console](/en/documentation/products/guides/getting-to-know-azion-console/) is the interface where you can access all the available products and their settings..

Reviewer: ```suggestion
- [Azion Console](/en/documentation/products/guides/getting-to-know-azion-console/) is the interface where you can access all the available products and their settings.
```

> - [Azion CLI](/en/documentation/products/azion-cli/overview/):  an open-source command-line interface (CLI) that allows interaction with Azion Edge Platform using a terminal. 

Reviewer: ```suggestion
- [Azion CLI](/en/documentation/products/azion-cli/overview/): an open-source command-line interface (CLI) that allows interaction with Azion Edge Platform using a terminal. 
```

- [Edge Application](/en/documentation/products/build/edge-application/): an Azion product that allows you to build applications that run in Azion's edge network.

> - [Edge Storage](/en/documentation/products/store/edge-storage/): a scalable and secure storage service designed to integrate object storage with the Azion Edge Computing Platform using the S3 standard for object operations.

Reviewer: ```suggestion
- [Edge Storage](/en/documentation/products/store/edge-storage/): a scalable and secure storage service designed to integrate object storage with the Azion Edge Platform using the S3 standard for object operations.
```

---

## Implementation 

- Initialize a Jamstack Edge Application based on templates:
[Framework specific guides]
- Link an existing application to an Azion edge application:
[Azion CLI link command]

---

## Related docs

> - [Azion Marketplace.](/en/documentation/products/marketplace/)
> - [Azion Edge Runtime Web APIs compatibility.](/en/documentation/devtools/runtime-apis/javascript/)
> - [Azion Edge Runtime Node.js compatibility.](/en/documentation/products/azion-edge-runtime/compatibility/node/)

Reviewer: ```suggestion
- [Azion Marketplace](/en/documentation/products/marketplace/)
- [Azion Edge Runtime Web APIs compatibility](/en/documentation/devtools/runtime-apis/javascript/)
- [Azion Edge Runtime Node.js compatibility](/en/documentation/products/azion-edge-runtime/compatibility/node/)
```
